lib/components/widgets/loginScreen/login_screen.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:classroom_app/components/widgets/CustomButton/custom_button.dart';
+import 'package:classroom_app/components/widgets/SignUpScreen/sign_up_screen.dart';
+
+class LoginScreen extends StatelessWidget {
+  const LoginScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+    final screenWidth = size.width;
+    final screenHeight = size.height;
+    return Scaffold(
+      backgroundColor: AppColors.white,
+      body: LayoutBuilder(
+        builder: (context, constraints) {
+          final isSmallScreen = constraints.maxHeight < 600;
+
+          return Padding(
+            padding: EdgeInsets.symmetric(horizontal: screenWidth * 0.05),
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                // Top spacing or logo section
+                if (!isSmallScreen)
+                  SizedBox(height: constraints.maxHeight * 0.1),
+                if (isSmallScreen) const Spacer(),
+                // Title
+                Text(
+                  'Log In',
+                  style: TextStyle(
+                    color: AppColors.homeScreenHeadingColor,
+                    fontSize: screenWidth * 0.08,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+                SizedBox(height: screenHeight * 0.02),
+                Text(
+                  'Enter your details below & free log in',
+                  style: TextStyle(
+                    color: AppColors.classroomSecondaryColor,
+                    fontSize: screenWidth * 0.045,
+                  ),
+                ),
+                SizedBox(height: screenHeight * 0.04),
+                // Input Fields
+                Expanded(
+                  flex: isSmallScreen ? 8 : 10,
+                  child: ListView(
+                    physics: isSmallScreen
+                        ? const BouncingScrollPhysics()
+                        : const NeverScrollableScrollPhysics(),
+                    padding: EdgeInsets.zero,
+                    children: [
+                      TextField(
+                        decoration: InputDecoration(
+                          labelText: 'Username',
+                          border: OutlineInputBorder(
+                            borderRadius: BorderRadius.circular(8),
+                          ),
+                        ),
+                      ),
+                      SizedBox(height: screenHeight * 0.02),
+                      TextField(
+                        obscureText: true,
+                        decoration: InputDecoration(
+                          labelText: 'Password',
+                          border: OutlineInputBorder(
+                            borderRadius: BorderRadius.circular(8),
+                          ),
+                        ),
+                      ),
```

Comment:
For managing the text fields, consider using TextEditingController for both username and password fields.
 ```
final _usernameController = TextEditingController();
final _passwordController = TextEditingController();

TextField(
  controller: _usernameController,
  // ...
);

TextField(
  controller: _passwordController,
  // ...
);
```